### PR TITLE
Disable provenance attestation to workaround docker buildx issue

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Disable provenance attestation to workaround docker buildx issue
related: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1712
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
For docker buildx build, set `--provenance=false` to workaround buildkit v0.11.0's enabling provenance attestation by default
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -145,7 +145,8 @@ build-ccm-image: buildx-setup docker-pull-prerequisites ## Build controller-mana
 		--build-arg ARCH="$(ARCH)" \
 		--build-arg VERSION="$(VERSION)" \
 		--file Dockerfile \
-		--tag $(CONTROLLER_MANAGER_IMAGE) .
+		--tag $(CONTROLLER_MANAGER_IMAGE) . \
+		--provenance=false
```
Comment (feiskyer) [inline Makefile:149]: similarly to provenance, could we set `--sbom=false` as well in case the default values change later?

Comment (lzhecheng) [reply Makefile]: Sure. Updated

Comment (andyzhangx) [reply Makefile]: so this workaround works?

Comment (lzhecheng) [reply Makefile]: We need to test in kubekins-e2e image so if this change does no harm then we can verify it in the pipeline.

Comment (andyzhangx) [reply Makefile]: ok, let's merge first.

**Comment by lzhecheng** [comment]
@andyzhangx this is not the PR with "hold" but needs lgtm :) instead. Thanks.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**